### PR TITLE
Refs [TASK][34790]Get data meal detail

### PR DESCRIPTION
**Pull request by GiangGiang1105**
## Related Tickets
- [#Task 34790: Get data meal detail](https://edu-redmine.sun-asterisk.vn/issues/34790)
## WHAT
- Get data meal detail
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/gfood/data/model/MealDetail.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    const val INGREDIENT1 = "strIngredient1"
+    const val INGREDIENT2 = "strIngredient2"
+    const val INGREDIENT3 = "strIngredient3"
+    const val INGREDIENT4 = "strIngredient4"
+    const val INGREDIENT5 = "strIngredient5"
+    const val INGREDIENT6 = "strIngredient6"
+    const val INGREDIENT7 = "strIngredient7"
+    const val INGREDIENT8 = "strIngredient8"
+    const val INGREDIENT9 = "strIngredient9"
+    const val INGREDIENT10 = "strIngredient10"
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/example/gfood/data/model/MealDetail.kt]: sao lại có vụ key số này e?

File: app/src/main/java/com/example/gfood/utils/Constant.kt [hunk 2]
```diff
@@ -4,5 +4,8 @@ object Constant {
     const val BASE_URL = "https://www.themealdb.com/api/json/v1/1/"
     const val BASE_CATEGORY = "categories.php"
     const val BASE_LIST_MEAL = "filter.php?c="
+    const val BASE_MEAL = "lookup.php?i="
+    const val BASE_URL_VIDEO = "https://www.youtube.com/watch?v="
     const val LIMIT_OFFSET = 3
+
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/example/gfood/utils/Constant.kt]: remove blank

File: app/src/main/java/com/example/gfood/data/source/remote/fetchjson/ParseJson.kt [hunk 3]
```diff
@@ -26,4 +24,34 @@ class ParseJson {
...
+        ingredients += "Ingredient: " + jsonObject?.getString(MealDetailEntry.INGREDIENT1) + ", " +
+                jsonObject?.getString(MealDetailEntry.INGREDIENT2) + ", " +
+                jsonObject?.getString(MealDetailEntry.INGREDIENT3) + ", " +
+                jsonObject?.getString(MealDetailEntry.INGREDIENT4) + ", " +
+                jsonObject?.getString(MealDetailEntry.INGREDIENT5) + ", " +
+                jsonObject?.getString(MealDetailEntry.INGREDIENT6) + ", " +
+                jsonObject?.getString(MealDetailEntry.INGREDIENT7) + ", " +
+                jsonObject?.getString(MealDetailEntry.INGREDIENT8) + ", " +
+                jsonObject?.getString(MealDetailEntry.INGREDIENT9) + ", " +
+                jsonObject?.getString(MealDetailEntry.INGREDIENT10)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/example/gfood/data/source/remote/fetchjson/ParseJson.kt]: parse json sao lại thế e?

File: app/src/main/java/com/example/gfood/data/model/MealDetail.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    const val LINK_VIDEO = "strYoutube"
+    const val INGREDIENT_ONE = "strIngredient1"
+    const val INGREDIENT_TWO = "strIngredient2"
+    const val INGREDIENT_THREE = "strIngredient3"
+    const val INGREDIENT_FOUR = "strIngredient4"
+    const val INGREDIENT_FIVE = "strIngredient5"
+    const val INGREDIENT_SIX = "strIngredient6"
+    const val INGREDIENT_SEVEN = "strIngredient7"
+    const val INGREDIENT_EIGHT = "strIngredient8"
+    const val INGREDIENT_NINE = "strIngredient9"
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/example/gfood/data/model/MealDetail.kt:33]: chụp ảnh đoạn json trả xuống a xem thử

Comment (GiangGiang1105) [reply app/src/main/java/com/example/gfood/data/model/MealDetail.kt:33]: image
ĐÂY HẢ ANH ?

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**